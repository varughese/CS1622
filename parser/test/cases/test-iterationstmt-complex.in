void main(void)
{
    int x;

    while(1 == 1)
        x + 1;

    while(1 == 1)
        1;

    while(1 == 1)
        (2);

    while(1 == 1);

    while(1 == 1)
    {
    }

    while(1 == 1)
        if(2 == 2);

    while(1 == 1)
        while(3 == 3);

    while(1 == 1)
        return;
}
