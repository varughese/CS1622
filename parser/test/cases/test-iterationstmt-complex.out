[program
  [fun-declaration
    [void]
    [main]
    [params]
    [compound-stmt 
      [var-declaration [int] [x]]
      [iteration-stmt 
        [==  [1] [1]]
        [+ [var [x]] [1]]
      ]
      [iteration-stmt 
        [==  [1] [1]] [1]
      ]
      [iteration-stmt 
        [==  [1] [1]] [2]
      ]
      [iteration-stmt 
        [==  [1] [1]]
        [;]
      ]
      [iteration-stmt 
        [==  [1] [1]]
        [compound-stmt 
        ]
      ]
      [iteration-stmt 
        [==  [1] [1]]
        [selection-stmt 
          [==  [2] [2]]
          [;]
        ]
      ]
      [iteration-stmt 
        [==  [1] [1]]
        [iteration-stmt 
          [==  [3] [3]]
          [;]
        ]
      ]
      [iteration-stmt 
        [==  [1] [1]]
        [return-stmt]
      ]
    ]
  ]
]
